### PR TITLE
docs: Add Storybook documentation landing page

### PR DESCRIPTION
**Pull request by JanHamara**
**What:**
Adds main landing page for the Storybook navigation (includes navigation to some of the most essential parts of the new documentation in SB)

**Why:**
We've put quite a lot of effort into migrating our old docs into Storybook. 

While Storybook is a great tool for documentation, isolated-development, testing **of components**, etc.  many top companies worldwide and their design systems use a combination of Storybook and a separate documentation website (that links to Storybook, Figma, repository source code, and more) to fulfil their documentation purposes. 

That could include:
- Page for component statuses (e.g. https://orbit.kiwi/component-status/)
- Changelogs
- Sharing docs for themeable design system, e.g. multiple color schemes (which will be case for us soon, due to rebranding)
- Design patterns docs
- How-to guides, migration guides, theming guides
- Library Hooks documentation
- Linking to Bug Reporting, Figma UI Kit, etc.
- Accessibility docs, Page for contributors community, Sandbox / Playground (not same thing as SB Canvas)

Some examples:
- https://mui.com/
- https://atlassian.design/
- https://primer.style/ (Github)
- https://chakra-ui.com/getting-started
- https://orbit.kiwi/foundation
- https://ux.mailchimp.com/

Anyways, **we don't currently have available time and effort to develop such sophisticated documentation website**, but this MR should be sort of an indicator of effort to integrate a bit of a more visual documentation and better look into our Storybook, on which we can develop later.
​
**How:**
- Uses Wave components to build up the welcome landing page on our Storybook documentation.
- Links to some of the most essential parts of docs for both newcomers and current users of the library (Colors, Typography, Spacing, Components, Icons, etc.).
- Provides a quick link to Bug Report page on Wave Github repo
- Provides a link to our Wave Figma file
- Highlights the ability to utilise the search functionality in SB with hotkey
​

**Media:**
Screenshot 2023-05-19 at 14 50 49

Screenshot 2023-05-19 at 14 51 30

**Checklist:**

-   [x] Review the design
-   [x] Ready to be merged

### CODE REVIEW

File: .storybook/manager.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// .storybook/manager.ts
+
+import { addons } from '@storybook/manager-api';
+import yourTheme from './YourTheme';
```
Comment (nlopin) [inline .storybook/manager.ts:4]: nit: rename to `FreenowTheme`?

File: docs/get-started.storybook.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import ByFreenow from './assets/by-freenow.svg';
+
+import { Box } from '../src/components/Box/Box';
+import { Card } from '../src/components/Card/Card';
+import { Headline } from '../src/components/Headline/Headline';
+import { Text } from '../src/components/Text/Text';
+
+<Meta title="Get Started" />
+
+{/* Logo */}
```
Comment (nlopin) [inline docs/get-started.storybook.mdx:12]: I'd consider removing this and the blank space below, if this does not break the layout

File: docs/get-started.storybook.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+You'll see this:
+
+<Card level={200}>
+    <Headline>Welcome</Headline>
+    <Text>FREE NOW Design System</Text>
+</Card>
+
+## Dynamic Imports & Create React App
```
Comment (nlopin) [inline docs/get-started.storybook.mdx:52]: Let's remove this piece, I don't think it is relevant anymore. It is [well supported](https://caniuse.com/es6-module-dynamic-import)

File: docs/index.storybook.mdx [hunk 4]
```diff
@@ -1,60 +1,298 @@
...
+    <img src={Map} className='hero-illustration' alt='map-illustration' />
 
-export default () => (
-    <Card level={200}>
-        <Headline>Welcome</Headline>
-        <Text>FREE NOW Design System</Text>
+  </Card>
+
+{' '}
+<Headline className="sb-unstyled" as="h3" mt="80px" mb="30px">
```
Comment (nlopin) [inline docs/index.storybook.mdx:144]: I see `sb-unstyled` applied in many places on this page. What if we wrap the entire content to `<Unstyled>` component which renders a div with `sb-unstyled` class?

File: .storybook/YourTheme.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// .storybook/YourTheme.ts
+import { create } from '@storybook/theming/create';
+import logo from '../public/assets/freenow-logo.svg';
+
+export default create({
+    base: 'light',
+
+    // Typography
+    fontBase: '"YOURNOWSans-Medium", "Open Sans", sans-serif',
```
Comment (nlopin) [inline .storybook/YourTheme.ts:9]: we need to serve the custom font, but I don't see it imported in the styled. I think we should bring the font file here and write `font-family` declaration

**Review by nlopin — CHANGES_REQUESTED** [review]
Great work Jan! It looks awesome 🏄‍♂️

File: .storybook/YourTheme.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    fontCode: 'monospace',
+
+    brandTitle: 'FREENOW',
+    brandUrl: 'https://www.free-now.com/',
```
Comment (jonotrujillo) [inline .storybook/YourTheme.ts:13]: This is the URL that the user is taken to when clicking on the FREENOW logo. IMO, it should be the URL of wave instead. What do you think?

File: .storybook/preview.ts [hunk 7]
```diff
@@ -65,7 +69,7 @@ export const preview: Preview = {
         options: {
             storySort: {
                 method: 'alphabetical',
```
Comment (jonotrujillo) [inline .storybook/preview.ts:71]: I think that when we have the `storySort.order` it's not needed to have a `storySort.method` as well

File: docs/assets/by-freenow.svg
Comment (jonotrujillo) [inline docs/assets/by-freenow.svg:1]: Is the serif "by" intentional. Looks a bit out of place since we don't use serif anywhere else (that I've noticed)

**Review by jonotrujillo — APPROVED** [review]

**Review by div-Leo — APPROVED** [review]

File: docs/assets/by-freenow.svg
Comment (JanHamara) [reply docs/assets/by-freenow.svg:1]: Hey Jonathan, where do you see "by" with serif? I can't see any 🤔

Comment (jonotrujillo) [reply docs/assets/by-freenow.svg:1]: Screenshot 2023-05-22 at 11 18 24

Can see it both in the diff in github but also when running the project locally.

File: .storybook/preview.ts (hunk 7 above)
Comment (JanHamara) [reply .storybook/preview.ts:71]: I'll check that, thanks 👍

File: docs/assets/by-freenow.svg
Comment (JanHamara) [reply docs/assets/by-freenow.svg:1]: Hmm, that's quite interesting. 🤔 

When I run it locally it is not in sans, also we don't have any fallback to serif font, but rather to **sans-serif**,  if it fell back from OpenSans (while in SVG it's set to 'YOURNOWSans-Medium')

Screenshot 2023-05-22 at 16 14 30

This should be fixed by adding the YOURNOW font, but anyways, quite weird situation 😅

File: .storybook/YourTheme.ts (hunk 6 above)
Comment (JanHamara) [reply .storybook/YourTheme.ts:13]: My way of thinking initially was that this is meant for brand URL as it's named, our lib being open-sourced would be a good way to link to company site, but I guess you're right, we can just use it as index page link. 🤔

File: docs/assets/freenow.svg
Comment (martimalek) [inline docs/assets/freenow.svg:1]: nit: should we call this file `freenow-logo.svg`? I see we use the word "logo" for others (Wave, Github, Figma)

File: docs/index.storybook.mdx [hunk 8]
```diff
@@ -1,60 +1,298 @@
...
+    }
+  `}
+</style>
 
-Welcome to the Design System of FREE NOW. Thanks for stopping by! This is the place where you find the latest components and essentials, that enable you to rapidly build your product.
+<img src={WaveBackground} className="wave-bg" width="100%" alt="wave-background" />
 
-## Installation
+<Box className='main-content'>
+  {/* Logo */}
```
Comment (martimalek) [inline docs/index.storybook.mdx:116]:
```suggestion
```

File: docs/index.storybook.mdx [hunk 9]
```diff
@@ -1,60 +1,298 @@
...
+        </Box>
+
+        <Text className='sb-unstyled' fontSize='14px' secondary>Convey meaning or structure, highlight various information or elements, display different states through our color palette.</Text>
+
+        <a href='/?path=/docs/essentials-colors--docs'>
+          <Button mt='24px' size='small' variant='secondary'>Explore &nbsp; <ArrowRightIcon /></Button>
+        </a>
+    </Card>
+
+    {/* TODO: Create Typography docs page - Example: https://orbit.kiwi/foundation/typography */}
```
Comment (martimalek) [inline docs/index.storybook.mdx:164]: Should we create an issue for this?

File: docs/index.storybook.mdx [hunk 10]
```diff
@@ -1,60 +1,298 @@
...
+        <Box>
+          <a href='/?path=/docs/essentials-icons-icons--docs'>
+            <Button mt='24px' size='small' variant='secondary'>Browse icons &nbsp; <ArrowRightIcon /></Button>
+          </a>
+        </Box>
+    </Card>
+
+  </Box>
+
+{/* Support */}
```
Comment (martimalek) [inline docs/index.storybook.mdx:240]:
```suggestion
```

File: docs/index.storybook.mdx [hunk 11]
```diff
@@ -1,60 +1,298 @@
...
-import { Card, Headline, Text } from '@freenow/wave';
+    <img src={Map} className='hero-illustration' alt='map-illustration' />
 
-export default () => (
-    <Card level={200}>
-        <Headline>Welcome</Headline>
-        <Text>FREE NOW Design System</Text>
+  </Card>
+
+{' '}
```
Comment (martimalek) [inline docs/index.storybook.mdx:143]: I see this repeated multiple times and I'm not sure what it's purpose is 😕

**Review by martimalek — APPROVED** [review]
Awesome work @JanHamara, it looks amazing!

File: docs/index.storybook.mdx (hunk 11 above)
Comment (JanHamara) [reply docs/index.storybook.mdx:143]: I think I had this auto-generated, will remove 😅

File: .storybook/preview.ts (hunk 7 above)
Comment (JanHamara) [reply .storybook/preview.ts:71]: indeed, looks like it doesn't

File: docs/index.storybook.mdx (hunk 4 above)
Comment (JanHamara) [reply docs/index.storybook.mdx:144]: yeah done that and removed className's with `sb-unstyled` class 👍

File: .storybook/YourTheme.ts (hunk 5 above)
Comment (JanHamara) [reply .storybook/YourTheme.ts]: Note: Adding `YOURNOWSans-Medium` as a font-family for theme was a mistake (due to me using it in Adobe AI, when generating the SVG for ''by Freenow" part in logo), it should be and will remain the default font of Wave DS which is `Open Sans` .

File: docs/assets/by-freenow.svg
Comment (JanHamara) [reply docs/assets/by-freenow.svg:1]: Fixed

**Pull request merged**